fabtests/benchmarks/benchmark_shared.c

```diff
@@ -248,28 +266,57 @@ int bandwidth(void)
 	return 0;
 }
 
-static int bw_rma_comp(enum ft_rma_opcodes rma_op)
+/**
+ * @brief Get completions of RMA operations, and verify_data if requested
+ *
+ * Completions are received for the rma_op, according to the sequence counter.
+ * Verifications assume <valid_windows> many operations of opts.transfer_size
+ * were completed starting at buf + offset_rma_start.
+ *
+ * @return 0 on success.
+*/
+static int bw_rma_comp(enum ft_rma_opcodes rma_op, int valid_windows)
```

Comment:
This call is the right place to handle data verification.  For RMA read operations, once all reads complete, we can just check that the data is correct.  For RMA writes, I see that you set FI_DELIVERY_COMPLETE below.  So, once the writes complete locally, the data should be visible at the peer.  We just need to send a message to the peer at this point for notification.  That is, the flow should be something like this:
```
complete all previously posted writes (opts.window_size completions)
send/recv sync message between peers (ft_sync)
verify write data
```